cupy/_core/core.pyx

```diff
@@ -2397,17 +2398,21 @@ cpdef _ndarray_base array(obj, dtype=None, bint copy=True, order='K',
     if hasattr(obj, '__cuda_array_interface__'):
         return _array_from_cuda_array_interface(
             obj, dtype, copy, order, subok, ndmin)
+
     if hasattr(obj, '__cupy_get_ndarray__'):
         return _array_from_cupy_ndarray(
             obj.__cupy_get_ndarray__(), dtype, copy, order, ndmin)
 
     concat_shape, concat_type, concat_dtype = (
         _array_info_from_nested_sequence(obj))
     if concat_shape is not None:
+        # TODO(leofang): handle blocking?
+        if blocking:
+            raise NotImplementedError
```

Comment:
I guess just synchronizing the stream here is enough?

Reply:
Done in 335aa8d9025332ed0b8358b13b7756135e7fb2d0, PTAL